### ISSUE TITLE
Introduce tooling support tests

### ISSUE THREAD

**Issue by sormuras** [junit-team/junit5#1429]
## Overview

With the cron-based daily build of the [junit5-samples](https://github.com/junit-team/junit5-samples) via [Travis CI](https://travis-ci.org/junit-team/junit5-samples/builds) some basic integration checks are in place. The sample projects use very simple setups on purpose and cover mainly the minimal "happy path" available.

The main JUnit 5 project should provide and maintain an "integration subproject" containing "end-to-end" checks using various build tools that natively support the JUnit Platform. The "integration subproject" will contain many standalone projects that will be executed as an "build integration test suite", and the respective build results will be verified.

See similar checks that Surefire performs at https://github.com/apache/maven-surefire/tree/master/surefire-its

## Deliverables

- [x] Create `platform-tooling-support-tests` project
- [x] Integrate `platform-tooling-support-tests` into build process -- as an opt-in build step
- [x] Support usage of local (SNAPSHOT) build artifacts

**Comment by sbrannen** [comment]
> Integrate "integration subproject" into build process -- decide if that subproject should be opt-in/out when "build" is triggered

I'd say it should definitely be disabled by default for developer builds (e.g., `gradle test`) but turned on as part of a nightly build or CI build.

**Comment by sbrannen** [comment]
If this project is limited to "build integration tests", I think we should pick a name that clearly expresses the scope -- for example, `platform-build-integration-tests` analogous to the existing `platform-tests` project.

**Comment by sbrannen** [comment]
Thanks for introducing the tooling support tests! 👍 

Looks like an elegant solution to the task at hand.

**Comment by sbrannen** [comment]
Reopening to address the following.

Javadoc is [published](https://junit.org/junit5/docs/snapshot/api/platform/tooling/support/package-summary.html) for the new `platform-tooling-support-tests` project, but it should not be.

```diff
 docs/snapshot/api/platform/tooling/support/Helper.html                    |  325 ++++++++++++
 docs/snapshot/api/platform/tooling/support/Request.Builder.html           |  429 ++++++++++++++++
 docs/snapshot/api/platform/tooling/support/Request.html                   |  432 ++++++++++++++++
 docs/snapshot/api/platform/tooling/support/Result.html                    |  409 ++++++++++++++++
 docs/snapshot/api/platform/tooling/support/Tool.Kind.html                 |  410 ++++++++++++++++
 docs/snapshot/api/platform/tooling/support/Tool.html                      |  557 +++++++++++++++++++++
 docs/snapshot/api/platform/tooling/support/class-use/Helper.html          |  163 ++++++
 docs/snapshot/api/platform/tooling/support/class-use/Request.Builder.html |  257 ++++++++++
 docs/snapshot/api/platform/tooling/support/class-use/Request.html         |  214 ++++++++
 docs/snapshot/api/platform/tooling/support/class-use/Result.html          |  209 ++++++++
 docs/snapshot/api/platform/tooling/support/class-use/Tool.Kind.html       |  224 +++++++++
 docs/snapshot/api/platform/tooling/support/class-use/Tool.html            |  240 +++++++++
 docs/snapshot/api/platform/tooling/support/package-frame.html             |   42 ++
 docs/snapshot/api/platform/tooling/support/package-summary.html           |  221 +++++++++
 docs/snapshot/api/platform/tooling/support/package-tree.html              |  196 ++++++++
 docs/snapshot/api/platform/tooling/support/package-use.html               |  216 ++++++++
```

**Comment by sbrannen** [comment]
Closing this issue since the JavaDoc publication issue has been addressed in 87221f8454d0e08144e445d25164640c80bdb52e.
